### PR TITLE
Add changeset generator

### PR DESCRIPTION
**Pull request by calebeby**
I made a little script that will diff the generated eslint config + plugin-rules, and output the changes to a changeset file as markdown format with links.

I used it to generate [this changeset](https://github.com/cloudfour/eslint-config/blob/4728476a98b99ed4a5041538e18132e5b2ef6175/.changeset/shaggy-coins-sell.md)

This is specifically intended to be used for dependency updates, where it isn't immediately obvious which rules changed, and it is tedious to figure out which rule changes actually get applied in this repo

### CODE REVIEW

**Review by Paul-Hebert — CHANGES_REQUESTED** [review]
The code looks good. I don't see the generated changeset you link though?

Could we add some documentation about this feature for future devs. Right now you've got to read through a decent amount of code to get context. I'd love if we could add a Markdown file explaining:

- What this script does
- Why it was written
- How to run it
- A short summary of how it works under the hood

File: generate-changeset.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Changeset Generator Script
+
+The changeset generator can be run using `node ./generate-changeset.mjs`. It will create a changeset containing the rule changes between the current branch and `main`.
+
+It was created to automate the tedious part of making changesets for this repo: Figuring out which rules changed, which rule changes are actually applicable, and creating/formatting markdown links.
+
+It works by cloning a second copy of the repo on `main`, and running the build both on the current branch, and on `main`, and then diffing the config and rules.
```
Comment (Paul-Hebert) [inline generate-changeset.md:7]: Perfect, thanks!

**Review by Paul-Hebert — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -33,7 +33,6 @@
     "@babel/core": "^7.13.10",
     "@typescript-eslint/eslint-plugin": "^4.0.0",
     "@typescript-eslint/parser": "^4.0.0",
-    "babel-eslint": "^10.0.1",
```
Comment (calebeby) [inline package.json:36]: This is unrelated but I just noticed that this was still here but we aren't using it

File: package.json [hunk 3]
```diff
@@ -23,7 +23,7 @@
     "!src/rules/**/*.test.js"
   ],
   "engines": {
-    "node": ">=8.0.0"
```
Comment (calebeby) [inline package.json:26]: This is not a breaking change because we already have many dependencies that only support node 12+

File: package.json [hunk 4]
```diff
@@ -60,9 +60,9 @@
     "typescript": "4.3.5"
   },
   "scripts": {
-    "check-lint": "eslint --format=pretty . && prettier --check .",
+    "check-lint": "eslint --format=pretty --ext=.js,.mjs,.cjs . && prettier --check .",
     "load-fixture-repo": "node fixtures/load-repo",
-    "lint": "eslint --format=pretty --fix . && prettier --write .",
+    "lint": "eslint --format=pretty --ext=.js,.mjs,.cjs --fix . && prettier --write .",
```
Comment (calebeby) [inline package.json:65]: This should be the default but ESLint has been really really slow to adopt this:

https://github.com/eslint/eslint/issues/9592
https://github.com/eslint/eslint/issues/13301
https://github.com/eslint/eslint/issues/14588

**Pull request merged**